files/justfiles/toggles.just

```diff
@@ -177,22 +177,84 @@ toggle-xwayland ACTION="prompt":
 
 # Toggle bash environment lockdown (mitigates LD_PRELOAD attacks)
 toggle-bash-environment-lockdown:
-    #!/usr/bin/bash
-    BASH_ENV_FILES=("$HOME/.bashrc" "$HOME/.bash_profile")
+    #!/bin/run0 /bin/bash
+    set -eou pipefail 
+    [ "$UID" -eq 0 ] || { echo "This script must be run as root."; exit 1;} #almost certainly not needed, but doesn't hurt
+    uid_min=$(grep -Po '^\s*UID_MIN\s+\K\d+' /etc/login.defs)
+    uid_max=$(grep -Po '^\s*UID_MAX\s+\K\d+' /etc/login.defs)
+    user_string=$(getent passwd | awk -F':' -v max="$uid_max" -v min="$uid_min" 'max >= $3 && $3 >= min {print $1}' | tr '\n' ',' | sed 's/,*$//')
+    IFS=',' read -ra user_list <<< "$user_string"
+    if lsattr "/etc/profile" 2>/dev/null | awk '{print $1}' | grep -q 'i'; then
+    	change="unlocked"
+    else
+    	change="locked"
+    fi
     echo "${b}WARNING${n} This will overwrite your .bashrc and .bash_profile."
     echo "This is needed to ensure the mitigation is effective."
+    echo ".bashrc, .bash_profile, and more will be ""${b}""$change""${n}""."
     echo "Do you understand?"
     echo "Please type in \"YES I UNDERSTAND\" and press enter"
-    read ACCEPT
-    if [ "$ACCEPT" == "YES I UNDERSTAND" ]; then
-      if lsattr "${BASH_ENV_FILES[0]}" 2>/dev/null | awk '{print $1}' | grep -q 'i'; then
-        echo "Bash environment '(${BASH_ENV_FILES[@]})' is locked down. Unlocking it."
-        for file in "${BASH_ENV_FILES[@]}"; do
-            run0 chattr -i "$file"
+    read -r ACCEPT
+    if [ "$ACCEPT" != "YES I UNDERSTAND" ]; then
+        exit
+    fi
+    echo "Do you want this change to apply to ""${b}""all""${n}"" users (""${user_list[@]}"")? [y/N]"
+    echo "Otherwise, it will only apply to the currently logged in user ($SUDO_USER)."
+    read -r REPLY
+    if [[ ! "$REPLY" == [yY]* ]]; then
+    	user_list=("$SUDO_USER")
+    fi
+    for user in "${user_list[@]}"; do
+    	echo "Working user:$user"
+    	user_home=$(getent passwd "$user" | awk -F':' '{ print $6}')
+    	BASH_ENV_FILES=("/etc/profile") #used because we know it exists, not that it needs to be immutable
+    	if [ "$change" == "locked" ]; then
+    		if [ -d "$user_home/.bashrc.d/" ]; then	
+    			echo "$user_home/.bashrc.d/ found" # performative, not needed
+        	else
+                mkdir -p "$user_home/.bashrc.d/" #will create folder if necessary
```

Comment:
Would you mind checking the indentation of every new `then`, `else`, `do` etc. blocks?